lib/OrganizationsFieldArray/OrganizationsFieldArray.js

```diff
@@ -0,0 +1,161 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import { uniq } from 'lodash';
+import { Field } from 'redux-form';
+
+import {
+  Button,
+  Col,
+  IconButton,
+  Layout,
+  Row,
+  Select,
+} from '@folio/stripes/components';
+
+import withKiwtFieldArray from '../withKiwtFieldArray';
+import { required } from '../validators';
+import OrganizationPicker from './OrganizationPicker';
+
+class OrganizationsFieldArray extends React.Component {
+  static propTypes = {
+    addContactBtnLabel: PropTypes.node,
+    change: PropTypes.func,
+    isEmptyMessage: PropTypes.node,
+    items: PropTypes.arrayOf(PropTypes.object),
+    name: PropTypes.string.isRequired,
+    onAddField: PropTypes.func.isRequired,
+    onDeleteField: PropTypes.func.isRequired,
+    roles: PropTypes.arrayOf(PropTypes.object),
+    organizationPickerComponent: PropTypes.elementType,
+  };
+
+  static defaultProps = {
+    addContactBtnLabel: <FormattedMessage id="stripes-erm-components.organizations.addOrganization" />,
+    isEmptyMessage: <FormattedMessage id="stripes-erm-components.organizations.noOrganizations" />,
+    organizationPickerComponent: OrganizationPicker,
+  }
+
+  state = {
+    organizations: {},
+  };
+
+  static getDerivedStateFromProps(props, state) {
+    if (props.items.length > Object.keys(state.organizations).length) {
+      const organizations = { state };
+      props.items.forEach(org => { organizations[org.id] = org; });
+      return { organizations };
+    }
+    return null;
+  }
+
+  validateDuplicateEntries = /* istanbul ignore next */ (value, allValues) => {
+    const orgs = allValues[this.props.name] || [];
+
+    const organizations = orgs.filter(organization => {
+      const { org } = organization;
+      return org ? org.name === value : [];
+    });
+    if (organizations.length < 2) return undefined;
+
+    const rolesForThisorg = orgs.map(org => org.role);
+    if (uniq(rolesForThisorg).length !== rolesForThisorg.length) {
+      return <FormattedMessage id="stripes-erm-components.organizations.noDuplicates" />;
+    }
+
+    return undefined;
+  }
+
+  addOrganization = /* istanbul ignore next */ (org, index) => {
+    const { name } = this.props;
+
+    this.setState(prevState => ({
+      organizations: {
+        ...prevState.organizations,
+        [org.id]: org,
+      }
+    }), () => this.props.change(`${name}[${index}].org.orgsUuid`, org.id)); // creates a new field in redux-form for orgsUuid
```

Comment:
I'm actually not in love with this approach anymore. I think another way of doing this, rather than having to carry `change` all the way down the component tree, would be to implement an `OrganizationField` component that basically renders/manages the components/object for the `${name}[${index}]`.

So that component's value would be something like `{ role: 'abc', org: { name: 'def', orgsUuid: 'ghi' } }`, and it can render controlled components directly (as opposed to `Field`s) and be able to easily manage all the callbacks/rendering.

I think that would actually end up being a lot cleaner in this component and wherever it's used.